Source/concepts/Concept.ts

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { IEquatableTo, IEquatable } from '@dolittle/rudiments';
+import { typeGuard } from '@dolittle/types';
+
+type ConceptBase = number | string | boolean | IEquatable;
```

Comment:
Just for completeness, I guess we could allow bigint as well? 
```suggestion
type ConceptBase = number | bigint | string | boolean | IEquatable;
```